### PR TITLE
Gradient checkpointing fix

### PR DESCRIPTION
**Pull request by michaelbenayoun**
# What does this PR do?

XLA-friendly gradient checkpointing function that accept keyword arguments.

### CODE REVIEW

File: optimum/neuron/models/training/training_utils.py [hunk 1]
```diff
@@ -199,3 +200,13 @@ def is_custom_modeling_model(model) -> bool:
...
+
+
+def checkpoint_with_kwargs(fn, *args, **kwargs):
+    """XLA-compatible gradient checkpointing that accepts keyword arguments via functools.partial."""
+    from functools import partial
+
+    from torch_xla.utils.checkpoint import checkpoint
+
+    fn_with_kwargs = partial(fn, **kwargs)
+    return checkpoint(fn_with_kwargs, *args)
```
Comment (tengomucho) [inline optimum/neuron/models/training/training_utils.py:212]: I think this should be the other way around:
```suggestion
    return checkpoint(*args, fn_with_kwargs)
```

Comment (michaelbenayoun) [reply optimum/neuron/models/training/training_utils.py:212]: No, the first argument must be the function we checkpoint, check [here](https://github.com/pytorch/xla/blob/master/torch_xla/utils/checkpoint.py#L219).

**Comment by michaelbenayoun** [comment]
> why do you need this function? Can you provide a test that shows its usage and tests that it works?

It is just a wrapper around the original checkpoint [function](https://github.com/pytorch/xla/blob/master/torch_xla/utils/checkpoint.py#L219). The goal is to make it possible to pass keyword arguments to the function we are wrapping with `checkpoint`.

I can add a test, but you mentioned many times that the CI was too long. For such a small change I was not considering it, please confirm you want a test.

File: optimum/neuron/models/training/llama/modeling_llama.py [hunk 2]
```diff
@@ -667,8 +668,8 @@ def forward(
         # Decoder layers
         for decoder_layer in self.layers[: self.config.num_hidden_layers]:
             if self.gradient_checkpointing and self.training:
-                hidden_states = checkpoint(
-                    decoder_layer.__call__,
+                hidden_states = checkpoint_with_kwargs(
```
Comment (tengomucho) [inline optimum/neuron/models/training/llama/modeling_llama.py:671]: you change this here, only for granite, but not for the other models. I think you should change it everywhere, or nowhere, or explain why you only changed it here.

Comment (michaelbenayoun) [reply optimum/neuron/models/training/llama/modeling_llama.py:671]: I dont undertsand? We change it "everywhere".

Comment (tengomucho) [reply optimum/neuron/models/training/llama/modeling_llama.py:671]: yep sorry, forgot that qwen3 inherited the forward from llama
